app/lib/actions/katello/content_view_puppet_environment/clone.rb

```diff
@@ -5,46 +5,63 @@ class Clone < Actions::Base
         attr_accessor :new_puppet_environment
 
         def plan(from_version, options)
-          environment = options[:environment]
-          new_version = options[:new_version]
+          environment = options.fetch(:environment, nil)
+          new_version = options.fetch(:new_version, nil)
+          puppet_modules_present = options.fetch(:puppet_modules_present, true)
           source = from_version.content_view_puppet_environments.archived.first
 
+          #don't create a cvpe if no puppet modules are present, but reuse it if it is present
           if environment
-            clone = find_or_build_puppet_env(from_version, environment)
+            clone = find_or_build_puppet_env(from_version, environment, puppet_modules_present)
+            return if clone.new_record? && !puppet_modules_present #not needed
```

Comment:
That end line comment confuses me everytime (I think that the line itself is not needed)